book/src/usage.md

```diff
@@ -60,5 +61,10 @@ Currently supported: `word`, `surround`.
 | ---                    | ---                      |
 | `w`                    | Word                     |
 | `(`, `[`, `'`, etc     | Specified surround pairs |
+| `f`                    | Function                 |
+| `c`                    | Class                    |
```

Comment:
Is this a good idea? What about languages without class?

Reply:
It's not classes per se, for example Rust doesn't have classes and it uses structs, enums, trait, etc for the class textobject.

Reply:
But how does user know if `f` or `c` is needed to select struct/enum/trait?

Reply:
The terms not being standardized across languages is a problem, but an advantage is that they are recognizable and intuitive. We also need a consistent set of keybindings across multiple languages. We could rename them into something more _technically_ correct, but then we would be sacrificing intuitiveness. I think what we can do here is to have better documentation describing what `function` and `class` means for each language:

| Language | `function` | `class` |
| --- | --- | --- |
| Rust | functions, methods | structs, enums, traits, union, impl block |
| Python | ... | ... |

This can serve as a reference in case there's confusion on their concrete meaning for a particular language, while still being intuitive for the majority of cases.

Reply:
Why not we do sort of expand region? It automatically selects the upper layer whether it is function or class?

Reply:
We actually do have an [`expand_selection`](https://github.com/helix-editor/helix/blob/05c2a72ccb7f79e8e581d2703816c74543d1995c/helix-term/src/commands.rs#L293) command that does exactly that. It's currently disabled (no default keybinding) since there is no `shrink_selection` counterpart yet. Also it selects the parent node of the current selected node, so you would have to potentially use `expand_selection` multiple times to select a function from inside its body, whereas the textobjects provide a quicker way to do it.